schema/cassandra/chardonnay/schema.cql

```diff
@@ -68,4 +68,32 @@ CREATE TABLE transactions (
     PRIMARY KEY  (transaction_id)
 ) WITH COMPACTION = {
      'class': 'org.apache.cassandra.db.compaction.LeveledCompactionStrategy'
-};
\ No newline at end of file
+};
+
+CREATE TYPE region (
+  name     text,
+  cloud    text
+);
+
+CREATE TYPE zone (
+  region   frozen<region>,
+  name     text
+);
+
+CREATE TYPE key_range (
+  base_range_uuid    uuid,
+  lower_bound_inclusive    blob,
+  upper_bound_exclusive    blob
+);
+
+CREATE TABLE keyspaces (
+    keyspace_id         uuid,
+    namespace           text,
+    name                text,
```

Comment:
we need to ensure that the <namespace,name> tuple is unique, so I would make that the key instead of the keyspace_id. When inserting, we must check that it doesn't already exist.